python/eos_TEST.py

```diff
@@ -163,6 +165,21 @@ def check_008_Model(self):
         except:
             raise TestFailedError('cannot determine running b quark mass')
 
+    """
+    Check if the latex strings of the Units are valid.
+    """
+    def check_009_Units(self):
+        from eos import Unit
+        import matplotlib
+        from matplotlib import texmanager
+
+        for attr, value in Unit.__dict__.items():
+            if attr[0] != "_" and attr != "latex":
+                try:
+                    matplotlib.texmanager.TexManager.get_text_width_height_descent(f"${value().latex()}$", 1)
```

Comment:
Is there a reason you can't just call value().latex() and catch errors? Without having to involve matplotlib? (I can't check it myself as for some reason for me the whole thing fails with a `TypeError: 'staticmethod' object is not callable` error)

Reply:
`value().latex()` only returns the latex string. So no TeX involved. If you want to check that the latex string is valid, you need to call a TeX parser. I found this to be the simplest ^^

Reply:
On Python 3.9, ``value()`` raises the type error mentioned by @MJKirk. The reason is that the method is a static method created by the equivalent of a ``@staticmethod`` decorat. E.g., the name ``Unit.Undefined`` refers to the decorator for the static method, not the method itself. On Python 3.9, we can circumvent the problem by using ``value.__func__()`` instead of ``value()``. Here's my proposed solution:
```diff
diff --git a/python/eos_TEST.py b/python/eos_TEST.py
index c7e939bc..e28107d9 100644
--- a/python/eos_TEST.py
+++ b/python/eos_TEST.py
@@ -176,7 +176,8 @@ class PythonTests:
         for attr, value in Unit.__dict__.items():
             if attr[0] != "_" and attr != "latex":
                 try:
-                    matplotlib.texmanager.TexManager.get_text_width_height_descent(f"${value().latex()}$", 1)
+                    s = value.__func__().latex()
+                    matplotlib.texmanager.TexManager.get_text_width_height_descent(f"${s}$", 1)
                 except:
                     raise TestFailedError('invalid latex string ' + value().latex())
 ```